File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<int> st;

        for(int i  = 0 ; i < s.length() ; i++){
            if (s[i] == '(' || s[i] == '[' || s[i] == '{') {
                st.push(s[i]);
            } 
            else {
                if(st.empty()) return 0;
                if(s[i] == ')' && st.top() != '(' || 
                    s[i] == ']' && st.top() != '[' ||
                    s[i] == '}' && st.top() != '{'){
                        return 0;
                    }
                st.pop();
            }
        }  

        return st.empty();

    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>() ;
        for(int i = 0 ; i < s.length() ; i++){
            char c = s.charAt(i) ;
            if(st.isEmpty()){
                if(c == ')' || c == '}' || c == ']')
                    return false ;
                else
                    st.push(c) ;
            }
            else if(st.peek() == '(' && c == ')' || st.peek() == '[' && c == ']' || st.peek() == '{' && c == '}')
                st.pop() ;
            else
                st.push(c) ;
        }
        if(!st.isEmpty())   return false ;
        return true ;
    }
}